modin/engines/base/frame/axis_partition.py

```diff
@@ -141,14 +141,30 @@ def apply(
             num_splits = len(self.list_of_blocks)
 
         if other_axis_partition is not None:
+            if not isinstance(other_axis_partition, list):
+                other_axis_partition = [other_axis_partition]
+            other_shape = (
+                len(other_axis_partition),
+                len(other_axis_partition[0].list_of_blocks),
+            )
+            if not self.axis:
+                other_shape = tuple(reversed(other_shape))
```

Comment:
Why is `reversed` needed?

Reply:
We need `other_shape` value to save the shape of broadcasting frame in terms of partitions before flattening it into 1D array. We want to see shape in that order `(nrows, ncols)`. We're computing `other_shape` basing on the length of `other_axis_partition` (array of axis partitions) and on the amount of partitions in axis partition (`list_of_blocks`).

Let's see now, how `other_shape` value depends on axis:

![image](https://user-images.githubusercontent.com/62142979/94056282-c48b7280-fde6-11ea-9de1-89b9431f2382.png)


![image](https://user-images.githubusercontent.com/62142979/94056263-bc333780-fde6-11ea-98a2-40e250645d91.png)